### ISSUE TITLE
Some code patterns cause `provideCodeLenses` to fail with an error

### ISSUE THREAD

**Issue by deadalusai** [firsttris/vscode-jest-runner#121]
We've hit some specific constructions which cause NREs in `provideCodeLenses`, which causes the code lenses for the whole file to disappear.

When using a "spec" pattern like the following:
```
describe("math", () => {
    const SPECS = [
        { name: "should work" },
    ];
    for (const spec of SPECS) {
        it(spec.name, () => { expect(1).toBe(1); });
    }
});
```

Scanning fails with the following exception:
```
[2020-11-17 10:49:43.127] [exthost] [error] TypeError: Cannot read property 'replace' of undefined
	at Object.t.escapeRegExp (c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:69:4102)
	at e (c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:396:2539622)
	at c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:396:2539495
	at Array.forEach (<anonymous>)
	at e (c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:396:2539473)
	at c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:396:2539840
	at Array.forEach (<anonymous>)
        at ...
```

One workaround is to wrap the spec name in a template string: ```it (`${spec.name}`, ...)```

This next one is weirder - invoking any function named `test` causes the scanner to fail:
```
describe("math", () => {
    const test = () => {};
    it("should work", () => { test(); });
});
```

Scanning fails with the following exception:
```
[2020-11-17 10:55:37.201] [exthost] [error] TypeError: Cannot read property 'text' of undefined
	at s (c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:381:26928)
	at c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:381:27638
	at u (c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:396:610937)
	at Object.d [as forEachChild] (c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:396:614755)
	at c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:381:27759
	at l (c:\Users\benjamin.fox\.vscode\extensions\firsttris.vscode-jest-runner-0.4.31\dist\extension.js:396:611021)
        at ...
```

Renaming the `test` function works around the issue.
![weird-code-lens-failure gif](https://user-images.githubusercontent.com/918952/99313451-749bca00-28c4-11eb-8c2d-6219930e1047.gif)

**Comment by deadalusai** [comment]
I wouldn't necessarily expect the test runner to understand these constructions, but I think it would be reasonable for the rest of `describe`/`it` statements in the file to have their code lenses available.

**Comment by firsttris** [comment]
thx for this detailed bug report

**Comment by vlechemin** [comment]
I have the same problem with this syntax:
```ts
test.each([
  'a',
  'b',
])(...)
```

**Comment by firsttris** [comment]
just linking this to this https://github.com/firsttris/vscode-jest-runner/issues/137

**Comment by the-michael-toy** [comment]
I don't know if this is helpful but I also see the labels disappear if my spec file ends with a function which calls describe and the first argument is an expression

```
function testFilesInDir(dirName: string) {
  describe("testing files in " + dirName, () => { });
}
testFilesInDir("a");
testFilesInDir("b");
```

Which is fixed by replace the call to describe with:

```
  describe(`testing files in ${dirName}`, () => { });
```

**Comment by firsttris** [comment]
should be fixed in v0.4.3.5 re-open if issue persists
